src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+    responses is a valid output format. Valid file formats are one of:
+    jsonl, xlsx, or csv.
+    Args:
+        file_formats (List[str]): A list of all of the file format strings
+                                  passed in by the user.
+    Returns:
+        None
+    """
+    for file_format in file_formats:
+        valid_formats = {item.value for item in IOFileType}
+        if file_format not in valid_formats:
+            raise ValueError(
+                f"File format {file_format} is not a valid output format. Format must be one of {valid_formats}"
+            )
+
+
+def create_results_file_name(
+    file_format: str, output_dir: str, timestamp: str, model_name: str
+) -> str:
+    """
+    Validates that file formats passed in for the file containing scores and
+    responses is a valid output format. Valid file formats are one of:
+    jsonl, xlsx, or csv.
+    Args:
+        file_formats (str): The file format for the output file. One of "csv",
+                            "xlsx", "jsonl".
+        output_dir (str):   Output directory for results and  scores file
+        timestamp (str):    timestamp in .iso format that is part of the
+                            name of the output file.
+        model_name (str):   Name of model to be used in the name of the output
+                            file.
+    Returns:
+        str: The file name containing the scores and model responses
+    """
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    elif IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+    elif IOFileType.JSONL.value == file_format:
+        file_type = "jsonl"
+    else:
+        raise ValueError("File format is not one of: csv, xlsx, jsonl")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.DK_BENCH_DATA_DIR
+        logger.debug(
+            "Output dir provided is not a directory or does not exist, writing results to default evaluation output directory %s",
+            output_dir,
+        )
+
+    return f"{output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+
+def make_run_dir(output_dir: str) -> str:
+    """
+    Creates a directory (run dir) under the user provided output directory that
+    contains all of the files with scores and responses for an innvocation of
+    `ilab model evaluate --benchmark dk_bench`.
+    Args:
+        output_dir (str):   Output directory that will contain the run dir.
+    Returns:
+        str: Name of run dir for result files to be written in.
+    """
+    timestamp = datetime.now().isoformat()
+
+    # remove any trailing slashes from user provided output_dir
+    output_dir = os.path.normpath(output_dir)
+    run_dir = f"{output_dir}/job-{timestamp}"
+
+    if not os.path.exists(output_dir) or not os.path.isdir(output_dir):
+        run_dir = f"{DEFAULTS.DK_BENCH_DATA_DIR}/job-{timestamp}"
```

Comment:
We should avoid writing to the `DEFAULTS` directories, what if a user changes where these directories are in their config?